Yasgui/packages/yasgui/src/TabElements.ts

```diff
@@ -367,4 +370,145 @@ export class TabList {
   }
 }
 
+class Editable extends HTMLInputElement {
+  edit? = false;
+}
+
+export class ExtendedTabListEl extends TabListEl {
+  private closeButton?: HTMLSpanElement;
+
+  public draw(name: string) {
+    this.tabEl = this.createTabRootElement();
+    this.renameEl = this.createRenameElement(name);
+    this.tabEl.prepend(this.renameEl);
+    this.closeButton = this.createCloseButton();
+    this.tabEl.appendChild(this.closeButton);
+    this.tabEl.appendChild(this.createLoaderElement());
+    return this.tabEl;
+  }
+
+  private createTabRootElement(): HTMLDivElement {
+    const tabEl = document.createElement("div");
+    tabEl.setAttribute("role", "presentation");
+    addClass(tabEl, "tab");
+    tabEl.addEventListener("keydown", (e: KeyboardEvent) => {
+      if (e.code === "Delete") {
+        this.handleDeleteTab();
+      }
+    });
+    return tabEl;
+  }
+
+  private valueChangedHandler(event: any) {
+    this.yasgui.getTab(this.tabId)?.setName(event.detail);
+  }
+
+  private componentModeChangedHandler(event: any): void {
+    // @ts-ignore
+    if (event.detail) {
+      addClass(this.closeButton, "hidden");
+    } else {
+      removeClass(this.closeButton, "hidden");
+    }
+  }
+
+  private renameElementOnBlurHandler(): void {
+    if (!this.tabEl) return;
+    if (this.tabEl.classList.contains("active")) {
+      this.renameEl?.setAttribute("tabindex", "0");
+    } else {
+      this.renameEl?.setAttribute("tabindex", "-1");
+    }
+  }
+
+  private renameElementOnFocusHandler(): void {
+    if (!this.tabEl) return;
+    if (this.tabEl.classList.contains("active")) {
+      this.tabList.tabEntryIndex = Object.keys(this.tabList._tabs).indexOf(this.tabId);
+    }
+  }
+
+  private renameElementOnClickHandler(e: MouseEvent): void {
+    e.preventDefault();
+    this.yasgui.selectTabId(this.tabId);
+  }
+
+  private createRenameElement(name: string): Editable {
+    const editable = document.createElement("ontotext-editable-text-field") as Editable;
```

Comment:
I like this approach because the renaming behavior is a separate functionality, and all the logic is contained within one component that can be reused. In the future, we can create a separate Stencil project, for example, 'Ontotext UI,' and move all components like this one into it. This project can be used wherever we need such components. Currently, we have:

AlertBox;
Dropdown;
Loader;
OntotextDialogWebComponent;
OntotextTooltipWebComponent;
Pagination.